packages/react/src/FilteredActionList/FilteredActionListWithModernActionList.tsx

```diff
@@ -98,64 +94,101 @@ export function FilteredActionList({
+          firstItem.onAction(firstItem, event)
+          event.preventDefault()
+        }
+      }
     },
-    [onListContainerRefChanged],
+    [items, groupMetadata],
   )
 
   useEffect(() => {
     onInputRefChanged?.(inputRef)
   }, [inputRef, onInputRefChanged])
 
-  useFocusZone(
-    {
-      containerRef: {current: listContainerElement},
-      bindKeys: FocusKeys.ArrowVertical | FocusKeys.PageUpDown,
-      focusOutBehavior: 'wrap',
-      focusableElementFilter: element => {
-        return !(element instanceof HTMLInputElement)
-      },
-      activeDescendantFocus: inputRef,
-      onActiveDescendantChanged: (current, previous, directlyActivated) => {
-        activeDescendantRef.current = current
-
-        if (current && scrollContainerRef.current && directlyActivated) {
-          scrollIntoView(current, scrollContainerRef.current, menuScrollMargins)
+  useEffect(() => {
+    if (items.length === 0) {
+      inputRef.current?.focus()
+    } else {
+      const itemIds = items.filter(item => item.selected).map(item => item.id)
+      const removedItem = selectedItems.find(item => !itemIds.includes(item))
+      if (removedItem && document.activeElement !== inputRef.current) {
+        const list = listRef.current
+        if (list) {
+          const firstSelectedItem = list.querySelector('[role="option"]') as HTMLElement
+          firstSelectedItem.focus()
         }
-      },
-    },
-    [
-      // List container isn't in the DOM while loading.  Need to re-bind focus zone when it changes.
-      listContainerElement,
-    ],
-  )
+      }
+    }
+  }, [items, inputRef, selectedItems])
 
+  // Focus management for active indicator line
   useEffect(() => {
-    // if items changed, we want to instantly move active descendant into view
-    if (activeDescendantRef.current && scrollContainerRef.current) {
-      scrollIntoView(activeDescendantRef.current, scrollContainerRef.current, {
-        ...menuScrollMargins,
-        behavior: 'auto',
-      })
+    const list = listRef.current
+    if (!list) return
+
+    const updateActiveIndicator = () => {
+      // Clear any existing active indicators
+      const activeItems = list.querySelectorAll(`.${classes.ActiveItem}`)
+      for (const item of activeItems) {
+        item.classList.remove(classes.ActiveItem)
+      }
+
+      if (inputRef.current && document.activeElement === inputRef.current) {
+        // When input is focused, mark the first item as active
+        const firstItem = list.querySelector('[role="option"]') as HTMLElement | null
+        if (firstItem) {
+          firstItem.classList.add(classes.ActiveItem)
+        }
+      } else if (list.contains(document.activeElement)) {
+        // When an item in the list is focused, mark it as active
+        const focusedItem = document.activeElement as HTMLElement
+        if (focusedItem.getAttribute('role') === 'option') {
+          focusedItem.classList.add(classes.ActiveItem)
+        }
+      }
```

Comment:
is there a world in which we can have css rules for this instead like:

- base case: if item is focused, then it has the style
- input focused case: add conditional data-input-focused attribute from javascript if inputRef.current && document.activeElement === inputRef.current to each element item and then on css side: if li:first-child[data-input-focused]

Reply:
@francinelucca I feel like this is a challenge and I want to accept but also I'm gonna approve your PR 😂